### PR TITLE
feat(unlock-app, unlock-js) - fix partial withdraw

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
`decimals` as 18 as default value and this is not correct 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Manage/elements/WithdrawFundModal.tsx [hunk 1]
```diff
@@ -84,6 +87,7 @@ export const WithdrawFundModal = ({
       lockAddress,
       beneficiary,
       amount: form.amount.toString(),
+      erc20Address: currencyContractAddress || ethers.constants.AddressZero,
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Manage/elements/WithdrawFundModal.tsx:90]:
```suggestion
      erc20Address: currencyContractAddress,
```

File: packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js [hunk 2]
```diff
@@ -12,12 +13,23 @@ import utils from '../../utils'
...
   callback
 ) {
   const lockContract = await this.getLockContract(lockAddress)
-  const tokenAddress = await lockContract.tokenAddress()
+  const tokenAddress = erc20Address || (await lockContract.tokenAddress())
+
+  // decimals could be 0!
+  if (decimals == null) {
+    // get the decimals from the ERC20 contract or default to 18
+    if (erc20Address !== ZERO) {
```
[the remark this answers is not in the document]
Comment (julien51) [reply packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js:26]: Let's make sure that `erc20Address` is not `null` to be more robust

File: packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js [hunk 3]
```diff
@@ -12,12 +13,23 @@ import utils from '../../utils'
...
  */
 export default async function (
-  { lockAddress, beneficiary, amount = '0', decimals = 18 },
+  { lockAddress, beneficiary, amount = '0', decimals, erc20Address },
   transactionOptions = {},
   callback
 ) {
   const lockContract = await this.getLockContract(lockAddress)
-  const tokenAddress = await lockContract.tokenAddress()
+  const tokenAddress = erc20Address || (await lockContract.tokenAddress())
```
Comment (julien51) [inline packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js:21]: You are not using `tokenAddress` below?

Comment (kalidiagne) [reply packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js:21]: the idea here is to retrieve `tokenAddress` when is not passed as prop

Comment (kalidiagne) [reply packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js:21]: oh i see what you mean

**Review by julien51 — APPROVED** [review]

**Pull request merged**